RFCs/polka-retrieval.md

```diff
@@ -0,0 +1,91 @@
+# Polka data retrieval
+
+This document aggregates research done on the topic of retrieving stored data
+from the storage providers. We didn't research the retrieval markets here and we
+assume that no additional incentives are needed for being part of the retrieval
+network. We assume that every storage provider also provides a retrieval
+service.
+
+### Use case
+
+We would like to support 3rd parties to retrieve data from the storage
+providers. To retrieve the data stored, they should provide CID (root node of
+the CAR file).
+
+## Retrieval providers indexer
+
+This is the service that Aidan is working on. The indexer will be responsible
+for holding the entire index of cids with corresponding retrieval providers. In
+this document we assume that the mapping between CID of the CAR file and
+retrieval providers is available to the retrieval clients.
+
+## Storage provider server
+
+The current server implementation exposes the endpoint which allows 3rd parties
+to upload some content. That content is then encoded as a CAR file. That files
+are then packed into the unsealed sector which is stored on disk. The unsealed
+sector is sealed and also stored on disk. The sealed sector is _unreadable_ that
+is why for a purposes of data retrieval we also need an unsealed sector. For the
+purposes of POC we expect that the unsealed sector is always available.
+
+### Local Index Directory
+
+As part of the storage server there is a need for a local index directory
+subsystem. The reason is because the unsealed sector doesn't contain any
+information about location of the CAR files. Without the indexer, the client
+would be able to retrieve only entire sectors. The indexer would allow us to
+target the retrieval of specific CAR file or even a specific blocks in the CAR
+file.
+
+More resources:
+
+- https://boost.filecoin.io/deployment/local-index-directory
+
+### Storage provider server - retrieval provider
+
+The process which provides retrievals should have access to the local index and
+also to the unsealed sectors data. So we expect the process to be running on the
+same machine or have access to the same storage device used by the provider. The
+local index is used to find the sector which contains the requested data and
+where exactly is that data located in the sector.
+
+Each retrieval provider spins of the p2p network used to share data with clients.
```

Comment:
This sentence doesn't make much sense. We'll discuss this off-GH & synchronously